src/lib/lib.jl

```diff
@@ -14,6 +14,7 @@ accum(x, y, zs...) = accum(accum(x, y), zs...)
 
 accum(x::Tuple, y::Tuple) = accum.(x, y)
 accum(x::AbstractArray, y::AbstractArray) = accum.(x, y)
+accum(x::DenseArray, y::AbstractArray) = x .= accum.(x, y)
```

Comment:
Test error might be relevant:
```
Error During Test at /var/lib/buildkite-agent/builds/p6000-gpuci3-julia-csail-mit-edu/julialang/zygote-dot-jl/test/cuda.jl:27
  | Got exception outside of a @test
  | DimensionMismatch("cannot broadcast array to have fewer dimensions")

[8] accum(x::Vector{Float32}, y::LinearAlgebra.Adjoint{Float32, Matrix{Float32}})
  | @ Zygote /var/lib/buildkite-agent/builds/p6000-gpuci3-julia-csail-mit-edu/julialang/zygote-dot-jl/src/lib/lib.jl:17
```
https://buildkite.com/julialang/zygote-dot-jl/builds/223#94c4e405-b3e9-4cff-8150-689a90dcf254

This seems to only happen on CuArrays. Maybe this is a bug that got fixed in Base, I forget?

Reply:
Gross. I feel like someone is playing loose and fast with their differential's types here.
I am guessing that somewhere a differnetial that should have been an `Adjoint{Float32, Vector{Float32}})` was instead 1x2  `Matrix{Float32}`.
And thus what should have happend was taking the adjoint of the adjoint of a vector  should have given us a `Vector`
(Since https://github.com/julialang/julia/issues/4774)
Anyway, if my theory is right then we should be able to trackdown and fix that root cause.

Otherwise we can special case this and call `accum(x::DenseVector, y::AbstractArray) = x .= accum.(x, vec(y))`
I think that code would always give the right answer or error?
as a hack, leaving a TODO linking to an issue to sort it out later?

Reply:
Yea 4774 is a pain sometimes. I think everything that can be, needs to be robust to this difference. I suppose CUDA hasn't caught up with Base here.